roles/cockpit-ui/tasks/main.yml

```diff
@@ -1,84 +1,35 @@
-    -n default
-  register: create_docker_registry_route
-  changed_when: "'already exists' not in create_docker_registry_route.stderr"
-  failed_when: "'already exists' not in create_docker_registry_route.stderr and create_docker_registry_route.rc != 0"
-  when: check_docker_registry_exists.rc == 0
-
-- name: Create passthrough route for registry-console
-  command: >
-    {{ openshift.common.client_binary }} create route passthrough
-    --service registry-console
-    --port registry-console
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
-  register: create_registry_console_route
-  changed_when: "'already exists' not in create_registry_console_route.stderr"
-  failed_when: "'already exists' not in create_registry_console_route.stderr and create_registry_console_route.rc != 0"
-  when: check_docker_registry_exists.rc == 0
-
-- name: Retrieve docker-registry route
-  command: >
-    {{ openshift.common.client_binary }} get route docker-registry
-    -o jsonpath='{.spec.host}'
-    --config={{ openshift_hosted_kubeconfig }}
-    -n default
+  oc_route:
+    kubeconfig: "{{ openshift_master_config_dir }}/admin.kubeconfig }}"
+    name: docker-registry
+    namespace: default
+    service_name: docker-registry
+    state: present
+    tls_termination: passthrough
   register: docker_registry_route
-  changed_when: false
-  when: check_docker_registry_exists.rc == 0
+  run_once: true
 
-- name: Retrieve cockpit kube url
-  command: >
-    {{ openshift.common.client_binary }} get route registry-console
-    -o jsonpath='https://{.spec.host}'
-    -n default
-  register: registry_console_cockpit_kube_url
-  changed_when: false
-  when: check_docker_registry_exists.rc == 0
+- name: Create passthrough route for registry-console
+  oc_route:
+    kubeconfig: "{{ openshift_master_config_dir }}/admin.kubeconfig }}"
+    name: registry-console
+    namespace: default
+    service_name: registry-console
+    state: present
+    tls_termination: passthrough
+  register: registry_console_cockpit_kube
+  run_once: true
 
 - name: Deploy registry-console
   command: >
     {{ openshift.common.client_binary }} new-app --template=registry-console
     {% if openshift_cockpit_deployer_prefix is defined  %}-p IMAGE_PREFIX="{{ openshift_cockpit_deployer_prefix }}"{% endif %}
     -p OPENSHIFT_OAUTH_PROVIDER_URL="{{ openshift.master.public_api_url }}"
-    -p REGISTRY_HOST="{{ docker_registry_route.stdout }}"
-    -p COCKPIT_KUBE_URL="{{ registry_console_cockpit_kube_url.stdout }}"
+    -p REGISTRY_HOST="{{ (docker_registry_route|from_json)['spec']['host'] }}"
+    -p COCKPIT_KUBE_URL="https://{{ (registry_console_cockpit_kube|from_json)['spec']['host'] }}"
     --config={{ openshift_hosted_kubeconfig }}
     -n default
   register: deploy_registry_console
   changed_when: "'already exists' not in deploy_registry_console.stderr"
   failed_when: "'already exists' not in deploy_registry_console.stderr and deploy_registry_console.rc != 0"
```

Comment:
I wonder if you could wrap all three tasks in a block and apply `run_once: true` there to avoid the duplication or potential breakage if the role is called from a play with multiple hosts.